ios/Classes/core/PHAsset+PM_COMMON.h

```diff
@@ -29,6 +29,7 @@ NS_ASSUME_NONNULL_BEGIN
 - (nullable NSString*)mimeType;
 - (BOOL)isAdjust;
 - (PHAssetResource *)getAdjustResource;
+- (PHAssetResource *)getUntouchedResource;
```

Comment:
This seems to affect the file of Live Photos. What's the difference between them? Any specs?

Reply:
_getAdjustResource_ return by default media with adjustments ( PHAssetResourceTypeFullSizePhoto/Video) if there are any.
_getUntouchedResource_ will always return original asset (PHAssetResourceTypePhoto/*Video).
LivePhotos are not affected by this, because there are already dedicated methods to deal with them. For example, when retrieving filename of an asset, will check if media type is LivePhoto and use the dedicated method:

```
- (NSString *)originalFilenameWithSubtype:(int)subtype {
    if (@available(iOS 9.1, *)) {
        if ([self isLivePhoto] && subtype == PHAssetMediaSubtypePhotoLive) {
            return [self getLivePhotosResource].originalFilename;
        }
    }
    PHAssetResource *resource = [self getUntouchedResource];
    if (resource) {
        return resource.originalFilename;
    }
```

_getLivePhotosResource_ currently will always returns original video asset ( PHAssetResourceTypePairedVideo), and there are no method to retrieve the asset with adjustments, hence we could add this possibility.

Reply:
Another thing I noticed, this code path is useless in 'getAdjustResource': 

```
- (PHAssetResource *)getAdjustResource {
    NSArray<PHAssetResource *> *resources = [PHAssetResource assetResourcesForAsset:self];
    if (resources.count == 0) {
        return nil;
    }
    
    if (resources.count == 1) {
        return resources[0];
    }
    
// // // useless, !isAdjust means resource count is one, 
// // // so will always be skipped by the previous check ( if resources.count == 1)

    if (![self isAdjust]) {
        for (PHAssetResource *res in resources) {
            if (self.mediaType == PHAssetMediaTypeImage
                && res.type == PHAssetResourceTypePhoto) {
                return res;
            }
            
            if (self.mediaType == PHAssetMediaTypeVideo
                && res.type == PHAssetResourceTypeVideo) {
                return res;
            }
        }
        
        return nil;
    }
    // // // 

...
```